### PR TITLE
fix: Return stateAction from sendOutboundAck

### LINKED ISSUE #418: Return stateAction from sendOutboundAck

### PR DESCRIPTION
**Pull request by fqutishat**
- return send functionality in func

closes #418

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/didcomm/protocol/didexchange/states.go [hunk 1]
```diff
@@ -255,14 +255,16 @@ func (s *completed) Execute(msg dispatcher.DIDCommMsg, thid string, ctx context)
 		}
 		return &noOp{}, action, nil
 	case ConnectionAck:
+		var action = func() error { return nil }
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/states.go]: action :=

File: pkg/didcomm/protocol/didexchange/states.go [hunk 2]
```diff
@@ -445,11 +447,13 @@ func (ctx *context) sendOutboundAck(msg dispatcher.DIDCommMsg) error {
 
 	err := json.Unmarshal(msg.Payload, ack)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	// TODO : Issue-353
 	sendVerKey := temp
-	return ctx.outboundDispatcher.Send(ack, sendVerKey, destination)
+	return func() error {
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/states.go]: Suggest assigning the func to a variable as a new line. (and then returning that variable.)

**Comment by troyronda** [comment]
@fqutishat a bit more description would be helpful :).

File: pkg/didcomm/protocol/didexchange/states.go (hunk 1 above)
Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/states.go]: done

File: pkg/didcomm/protocol/didexchange/states.go (hunk 2 above)
Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/states.go]: done

**Pull request merged**